### PR TITLE
Add content holder

### PR DESCRIPTION
**Pull request by AlexBalo**

### CODE REVIEW

File: dialogplus/src/main/java/com/orhanobut/dialogplus/ViewHolder.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            }
+        });
+        addContent(inflater, parent, contentContainer);
+        headerContainer = (ViewGroup) view.findViewById(R.id.header_container);
+        footerContainer = (ViewGroup) view.findViewById(R.id.footer_container);
+        return view;
+    }
+
+    private void addContent(LayoutInflater inflater, ViewGroup parent, ViewGroup container) {
+        if (viewResourceId != -1) {
```
Comment (orhanobut) [inline dialogplus/src/main/java/com/orhanobut/dialogplus/ViewHolder.java]: magic number :) we can use INVALID here

File: dialogplus/src/main/java/com/orhanobut/dialogplus/DialogPlus.java [hunk 2]
```diff
@@ -338,25 +338,29 @@ private View createView(LayoutInflater inflater) {
...
         holder.addHeader(headerView);
 
         loopViewsRecursively(footerView);
         holder.addFooter(footerView);
 
-        if (adapter != null) {
-            holder.setAdapter(adapter);
-        }
+        if (adapter != null && holder instanceof HolderAdapter) {
+            ((HolderAdapter) holder).setAdapter(adapter);
```
Comment (orhanobut) [inline dialogplus/src/main/java/com/orhanobut/dialogplus/DialogPlus.java]: local variable would be nice here. We can avoid one extra casting
<pre>
HolderAdapter holderAdapter = (HolderAdapter) adapter;
</pre>

File: dialogplus/src/main/java/com/orhanobut/dialogplus/DialogPlus.java [hunk 3]
```diff
@@ -338,25 +338,29 @@ private View createView(LayoutInflater inflater) {
         holder.setBackgroundColor(backgroundColorResourceId);
         View view = holder.getView(inflater, rootView);
 
+        if (holder instanceof ViewHolder) {
+            loopViewsRecursively(view);
+        }
+
         loopViewsRecursively(headerView);
```
Comment (orhanobut) [inline dialogplus/src/main/java/com/orhanobut/dialogplus/DialogPlus.java]: what do you think if we rename this method? like initClickListenerRecursively(view) to make it more clear?

**Comment by orhanobut** [comment]
we can also remove simplelistview as well, since we don't use it anymore.

File: dialogplus/src/main/java/com/orhanobut/dialogplus/ViewHolder.java (hunk 1 above)
Comment (AlexBalo) [inline dialogplus/src/main/java/com/orhanobut/dialogplus/ViewHolder.java]: Yeah, sure. I declared it on top and then forgot to use it

File: dialogplus/src/main/java/com/orhanobut/dialogplus/DialogPlus.java (hunk 2 above)
Comment (AlexBalo) [inline dialogplus/src/main/java/com/orhanobut/dialogplus/DialogPlus.java]: Definitely, I agree

File: dialogplus/src/main/java/com/orhanobut/dialogplus/DialogPlus.java (hunk 3 above)
Comment (AlexBalo) [inline dialogplus/src/main/java/com/orhanobut/dialogplus/DialogPlus.java]: Maybe assignClickListenerRecursively(view)? What do you think?

**Comment by AlexBalo** [comment]
Yes, I will remove the dependency

File: dialogplus/src/main/java/com/orhanobut/dialogplus/DialogPlus.java (hunk 3 above)
Comment (orhanobut) [inline dialogplus/src/main/java/com/orhanobut/dialogplus/DialogPlus.java]: much better indeed

**Comment by AlexBalo** [comment]
Fixing comments from review

**Comment by AlexBalo** [comment]
I pushed the javadoc for the DialogPlus class

**Comment by orhanobut** [comment]
LGTM

**Pull request merged**